New mod type: 'gadget' (a self-contained script)

ohlidalp opened the pull request:
![image](https://github.com/user-attachments/assets/66202795-2acc-4606-8d91-cec03fc9428d)


Gadgets are essentially scripts wrapped as modcache entries. The point is to be able to distribute community-created scripts via the repository. The name "gadgets" is inspired by Beyond All Reason RTS (running open source Recoil engine). Usual modcache treatment applies:
* can be zipped or unzipped in directory
* can have preview (aka 'mini') image
* selectable via MainSelectorUI (open from TopMenubarUI, menu Tools, button BrowseGadgets

All methods of loading a script were extended to recognize the .gadget suffix and load the gadget appropriately:
* the 'loadscript' console command
* the cvars 'app_custom_scripts' and 'app_recent_scripts'
* the -runscript command line argument
NOTE that although `ScriptCategory::GADGET` was added, all the above methods still load everything as `ScriptCategory::CUSTOM` and the game adjusts it by detecting the '.gadget' file extension.

The ScriptMonitorUI was extended to display .gadget file name for `ScriptCategory::GADGET` script units.

Example .gadget file:
[enginetool-gadget.zip](https://github.com/user-attachments/files/18926435/enginetool-gadget.zip)

```
; This is a .gadget mod - basically a script wrapped as a modcache entry.
; The .gadget file must be present (even if empty) to get recognized.
; The script must have same base filename, i.e. 'foo.gadget' -> 'foo.as'
; Any extra include scripts or other resources can be bundled.
; -----------------------------------------------------------------------

; Name to display in Selector UI.
gadget_name "Engine Tool"

; Authors (Syntax: <credit>, <forumID>, <name>, [<email>]) - multiple authors can be given.
gadget_author "base script" 351 ohlidalp

; Description to display in Selector UI.
gadget_description "In-game engine diag and adjustment."

; Category for Selector UI (300 = Generic gadgets, 301 = Actor gadgets, 302 = Terrain gadgets).
gadget_category 301
```

ohlidalp commented:
I've converted the bundled editor scripts to gadgets.

![image](https://github.com/user-attachments/assets/5da960fb-71df-487e-a23f-52a81d8955b5)

This should make the mature tools (script editor, road editor) a lot more accessible and advertised via TopMenubar/Tools menu/[Browse gadgets...] button.
Most of the scripts are alpha prototypes (previously example_*.as scripts), but they're something to show anyway.

CuriousMike56 commented:
Received these warnings on first terrain load:
![RoR_2025-02-26_23-04-32](https://github.com/user-attachments/assets/413fa9b1-b575-4127-bc9b-dac7a2c99d1e)
Everything appears to work OK otherwise.

ohlidalp commented:
> @CuriousMike56 Received these warnings on first terrain load:

Fixed. Those files had newlines in description strings, which is illegal. I've updated the 'gadget_description' directive: it can now be used multiple times to create multiline description.
